File: Code/Avernakis Nodejs/Test-Nodejs/src/Ave/Ui/Visual/UiLabel.ts
import { AveLib } from "../../AveLib";
import { AlignType } from "../UiCommon";
import { Vec4 } from "../../Math/Vector";
import { IVisual } from "../UiVisual";
import { Window } from "../Control/UiWindow";

export interface ILabel extends IVisual {
    new (window: Window, key?: string | number): ILabel;

    SetText(s: string): Label;
    GetText(): string;

    SetAlignHorz(n: AlignType): Label;
    GetAlignHorz(): AlignType;

    SetAlignVert(n: AlignType): Label;
    GetAlignVert(): AlignType;

    SetBackColor(vColor: Vec4): Label;
    GetBackColor(): Vec4;

    SetWrappable(b: boolean): Label;
    GetWrappable(): boolean;
}

export class Label extends (AveLib.UiLabel as ILabel) {}

File: Code/Avernakis Nodejs/Avernakis-Nodejs/Private/UiLabel.cpp
#include "StdAfx.h"
#include "UiLabel.h"

#define ThisMethod($x) &UiLabel::$x
#define AutoAddMethod($x, ...) AddMethod<__VA_ARGS__>( #$x, ThisMethod( $x ) )
#define MakeThisFunc($x) MakeFunc( this, &UiLabel::$x )

namespace Nav
{

	namespace
	{
		ObjectRegister<UiLabel> c_obj;
	}

	void UiLabel::DefineControl()
	{
		AutoAddMethod( SetText );
		AutoAddMethod( GetText, WrapObjectGeneric );
		AutoAddMethod( SetAlignHorz );
		AutoAddMethod( GetAlignHorz, WrapObjectGeneric );
		AutoAddMethod( SetAlignVert );
		AutoAddMethod( GetAlignVert, WrapObjectGeneric );
		AutoAddMethod( SetBackColor );
		AutoAddMethod( GetBackColor, WrapObjectGeneric );
		AutoAddMethod( SetWrappable );
		AutoAddMethod( GetWrappable, WrapObjectGeneric );
	}

	U1 UiLabel::Ctor( UiWindow * p, Napi::Value v )
	{
		if ( !__CreateControl( p, v ) )
			return false;

		return true;
	}
}
